langchain/document_loaders/audio.py

```diff
@@ -0,0 +1,35 @@
+import os
+import openai
+from typing import List
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+class AudioFileLoader(BaseLoader):
+    """Document loader for audio files using audio-to-text transcription with OpenAI Whisper model.
+    
+    Example:
+    .. code-block:: python
+        from langchain.document_loaders import AudioFileLoader
+        audio_file_path = "/path/to/directory"
+        loader = AudioFileLoader(audio_file_path)
+        loader.load()
+    """
+
+    def __init__(self, audio_file_path: str = "text"):
+        """Initialize with path to audio file. 
+
+        Args:
+            audio_file_path: Path to directory to load from
+        """
+        self.audio_file_path = audio_file_path
+
+    def lazy_load(self) -> Document:
+        """Transcribe audio file to text w/ OpenAI Whisper API."""
+        audio_file = open(self.audio_file_path , "rb")
```

Comment:
Generally for opening files, it's best o always include a context manager to no file descriptors remain open
```suggestion
        with open(self.audio_file_path, 'rb') as f:
           audio_file = f.read()
```

Reply:
Minor nit on this: the `bytes` object is created by calling `f.read()` inside a with statement doesn't have a name attribute associated with it, which throws an error w/ the Whisper API. But, we can work around it by explicitly adding name (e.g., audio_file.name = self.audio_file_path).